packages/context/src/binding.ts

```diff
@@ -216,8 +225,16 @@ export class Binding {
    * ```
    */
   getValue(ctx: Context): BoundValue | Promise<BoundValue> {
-    // First check cached value for non-transient
+    // Make sure we're not endlessly looping over our subcontext.
+    if (this.subcontext && this.subcontext !== ctx) {
```

Comment:
We probably need to create the subContext per `getValue()` call to avoid racing condition. Theoretically speaking, the same binding instance can be used against multiple instances of `Context`. Switching the parent on demand is not safe.

Reply:
If we were to make a new context on every getValue call, we would need to:
- Store the value(s) to bind "elsewhere".
- Generate a new context each request.
- Populate that context with the stored data.

As it currently is now, you store the data in the new orphaned Context immediately which means:
- Only one context is created per overriding binding, rather than one per request
- No intermediate structure is required to store the data above and beyond the binding's context instance.
- Subsequent requests will not have to prepare the context with data from that intermediate container.

Reply:
At this moment, a binding is not owned by a context. For example, we can do:
```
const binding  = new Binding('a');
const ctx1 = new Context();
const ctx2 = new Context();

binding.getValue(ctx1);
binding.getValue(ctx2);
```
This will cause racing condition if you store `subContext` with the binding and switch parent upon each call of `getValue()`.

Reply:
Is this still true with the circular linking check I've added to `.setParent`?